### PR TITLE
feat(packaging) improve & automate packaging

### PR DESCRIPTION
**Pull request by jmaupetit**
- [x] improve setup scripts and configuration
- [x] add packaging workflow with CircleCI & GemFury
- [x] update install documentation

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,164 @@
+# fun packaging CI
+version: 2
+
+# List jobs that will be integrated and executed in a workflow
+jobs:
+
+  # Build fun application package
```
Comment (sampaccoud) [inline .circleci/config.yml]: copy/paste fail?

Build xblock package?

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+
+      # Cache virtual environment
+      - save_cache:
+          paths:
+            - ./venv
+          key: v1-dependencies-{{ checksum "setup.cfg" }}
+
+      # Build the package (source + wheel binary)
+      - run:
+          name: Build fun application package
```
Comment (sampaccoud) [inline .circleci/config.yml]: Build xblock package?

File: .circleci/config.yml [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+            echo ":::: Source distribution ::::"
+            tar tvzf dist/*.tar.gz
+
+      # Cache built distributions
+      - save_cache:
+          paths:
+            - ./dist
+          key: v1-distributions-{{ checksum "setup.cfg" }}
+
+  # Check that target git tag refer to a master branch commit
```
Comment (sampaccoud) [inline .circleci/config.yml]: ... and corresponds to the version in setup.cfg

**Review by sampaccoud — APPROVED** [review]
Some minor comments but LGTM

File: .circleci/config.yml (hunk 1 above)
Comment (jmaupetit) [reply .circleci/config.yml]: No i tried to make configuration generic. It's a FUN application. WDYT?

File: .circleci/config.yml (hunk 2 above)
Comment (jmaupetit) [reply .circleci/config.yml]: Same thing here.

File: .circleci/config.yml (hunk 1 above)
Comment (sampaccoud) [reply .circleci/config.yml]: Maybe here just "Build application package" is better.

**Pull request merged**